internal/jimm/juju/migrationtarget_test.go

```diff
@@ -239,11 +243,107 @@ func TestPrechecks_ModifiesModelDescription(t *testing.T) {
 	dbUser := env.User("alice@canonical.com").DBObject(c, j.Database)
 	user := openfga.NewUser(&dbUser, nil)
 
-	model := newMigrationInfo("bob")
+	model := newMigrationInfo(modelDescriptionArgs{
+		Owner:               "bob",
+		ModelName:           "test-model",
+		CloudName:           "test",
+		CloudCredentialName: "test-cred",
+		CloudRegionName:     "test-region",
+	})
 	err := j.Prechecks(ctx, user, model)
 	c.Assert(err, qt.IsNil)
 }
 
+func TestPrechecks_MissingCloudRegion(t *testing.T) {
+	c := qt.New(t)
+	ctx := context.Background()
+
+	api := &jimmtest.API{}
+
+	j := newTestJujuManager(c, &parameters{
+		Dialer: &jimmtest.Dialer{
+			API: api,
+		},
+	})
+
+	env := jimmtest.ParseEnvironment(c, testEnvWithIncomingMigration)
+	env.PopulateDBAndPermissions(c, j.ResourceTag(), j.Database, j.OpenFGAClient)
+
+	dbUser := env.User("alice@canonical.com").DBObject(c, j.Database)
+	user := openfga.NewUser(&dbUser, nil)
+
+	model := newMigrationInfo(modelDescriptionArgs{
+		Owner:               "bob",
+		ModelName:           "test-model-2",
+		CloudName:           "test",
+		CloudCredentialName: "test-cred",
+		CloudRegionName:     "test-region-not-found",
+	})
+
+	err := j.Prechecks(ctx, user, model)
+	c.Assert(err, qt.ErrorMatches, `^failed to find region for cloud "test".*`)
+}
+
+func TestPrechecks_MissingCloud(t *testing.T) {
+	c := qt.New(t)
+	ctx := context.Background()
+
+	api := &jimmtest.API{}
+
+	j := newTestJujuManager(c, &parameters{
+		Dialer: &jimmtest.Dialer{
+			API: api,
+		},
+	})
+
+	env := jimmtest.ParseEnvironment(c, testEnvWithIncomingMigration)
+	env.PopulateDBAndPermissions(c, j.ResourceTag(), j.Database, j.OpenFGAClient)
+
+	dbUser := env.User("alice@canonical.com").DBObject(c, j.Database)
+	user := openfga.NewUser(&dbUser, nil)
+
+	model := newMigrationInfo(modelDescriptionArgs{
+		Owner:               "bob",
+		ModelName:           "test-model-2",
+		CloudName:           "test-not-found",
+		CloudCredentialName: "test-cred",
+		CloudRegionName:     "test-region",
+	})
+
+	err := j.Prechecks(ctx, user, model)
+	c.Assert(err, qt.ErrorMatches, `^failed to find region for cloud "test-not-found".*`)
```

Comment:
Can we test the rest of this error explicitly? I was expecting to see ""test-not-found": cloud doesn't exist" or something.